### ISSUE TITLE
[DOCS] The gradient issue with no error reminder when passing torch tensor into kernal in differentiable setting

### ISSUE THREAD

**Issue by Jianghanxiao** [NVIDIA/warp#394]
### Bug Description

![Image](https://github.com/user-attachments/assets/03a665d5-0874-4889-a091-75f303104002)
Following the document, it's a bit misleading. At the start, I think "cannot handle" means something like "requires_grad=False" for the torch tensor passing to the kernel. However, after constructing my own stuffs, I find it surprisingly modify the original value of the torch tensor. Now I understand that "cannot handle" means that if using this way in a differentiable setting, there will definitely be a bug. But there should be some error message for this if such operations are weird.

Here is a minimum test example to show the above problems

```python
import torch
import warp as wp


@wp.kernel
def test_kernel(
    x: wp.array(dtype=wp.vec3), y: wp.array(dtype=wp.vec3), z: wp.array(dtype=wp.vec3)
):
    tid = wp.tid()
    z[tid] = x[tid] + y[tid]


wp.init()
wp.set_device("cuda:0")

x = torch.ones((10, 3), dtype=torch.float32, device="cuda")
y = torch.ones((10, 3), dtype=torch.float32, device="cuda")
wp_y = wp.from_torch(y, dtype=wp.vec3, requires_grad=True)
z = torch.zeros((10, 3), dtype=torch.float32, device="cuda")
wp_z = wp.from_torch(y, dtype=wp.vec3, requires_grad=True)

tape = wp.Tape()

with tape:
    wp.launch(test_kernel, dim=10, inputs=[x, wp_y], outputs=[wp_z])

print(x) # Output all 1

tape.backward(grads={wp_z: wp.ones_like(wp_z)})

print(x) # Output all 2
``` 

And here is the output
![Image](https://github.com/user-attachments/assets/dc4abd69-7902-475c-9382-aee4701c53ad)

Hope this can help others who misunderstand the document and don't have any reported error messages or warnings

**Comment by daedalus5** [comment]
Thanks for pointing this out. We have a fix ready for the next release.
